chainer/functions/normalization/batch_normalization.py

```diff
@@ -97,18 +99,24 @@ def backward(self, axis, gamma, gy, x, xp,
             # convert it to numpy here.
             gy = numpy.asarray(gy)
 
-        x_hat = _x_hat(x, mean[expander], inv_std[expander])
+        x_hat = _x_hat(x, mean[expander],
+                       inv_std[expander]).astype(x.dtype, copy=False)
         gbeta, ggamma = self.get_ggamma_and_gbeta(axis, gamma, gy, x_hat, xp)
 
         inv_m = gamma.dtype.type(1. / (x.size // gamma.size))
         if xp is numpy:
+            # cast to avoid a error of "mkldnn::error"
+            if gy.dtype == numpy.float64 and gamma.dtype == numpy.float32:
+                gamma = gamma.astype(numpy.float64, copy=False)
```

Comment:
It is not elegant to consider ideep in the generic implementation.
I'm going to refator this part later.

Reply:
I thought I could replicated the error (actually segfault) yesterday but not now...

Reply:
OK, replicated in this PR branch, but not in master...

Reply:
#8206